### ISSUE TITLE
role_definition_id attribute for data source azurerm_role_definition is incorrect

### ISSUE THREAD

**Issue by BzSpi** [hashicorp/terraform-provider-azurerm#11172]
The data source `azurerm_role_definition` outputs the role id instead of the role guid in the attribute `role_definition_id`.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform 0.14 & Provider azurem 2.53

### Affected Resource(s)

* `data.azurerm_role_definition`

### Terraform Configuration Files

```hcl
data "azurerm_role_definition" "builtin" {
  name = "Contributor"
}
```

### Expected Behaviour

```
> data.azurerm_role_definition.builtin
{
  "assignable_scopes" = tolist([
    "/",
  ])
  "description" = "Grants full access to manage all resources, but does not allow you to assign roles in Azure RBAC, manage assignments in Azure Blueprints, or share image galleries."
  "id" = "/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c"
  "name" = "Contributor"
  "permissions" = tolist([
    {
      "actions" = tolist([
        "*",
      ])
      "data_actions" = toset([])
      "not_actions" = tolist([
        "Microsoft.Authorization/*/Delete",
        "Microsoft.Authorization/*/Write",
        "Microsoft.Authorization/elevateAccess/Action",
        "Microsoft.Blueprint/blueprintAssignments/write",
        "Microsoft.Blueprint/blueprintAssignments/delete",
        "Microsoft.Compute/galleries/share/action",
      ])
      "not_data_actions" = toset([])
    },
  ])
  "role_definition_id" = "b24988ac-6180-42a0-ab88-20f7382dd24c"
  "scope" = tostring(null)
  "timeouts" = null /* object */
  "type" = "BuiltInRole"
}
```

### Actual Behaviour

```
> data.azurerm_role_definition.builtin
{
  "assignable_scopes" = tolist([
    "/",
  ])
  "description" = "Grants full access to manage all resources, but does not allow you to assign roles in Azure RBAC, manage assignments in Azure Blueprints, or share image galleries."
  "id" = "/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c"
  "name" = "Contributor"
  "permissions" = tolist([
    {
      "actions" = tolist([
        "*",
      ])
      "data_actions" = toset([])
      "not_actions" = tolist([
        "Microsoft.Authorization/*/Delete",
        "Microsoft.Authorization/*/Write",
        "Microsoft.Authorization/elevateAccess/Action",
        "Microsoft.Blueprint/blueprintAssignments/write",
        "Microsoft.Blueprint/blueprintAssignments/delete",
        "Microsoft.Compute/galleries/share/action",
      ])
      "not_data_actions" = toset([])
    },
  ])
  "role_definition_id" = "/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c"
  "scope" = tostring(null)
  "timeouts" = null /* object */
  "type" = "BuiltInRole"
}
```

### Steps to Reproduce

1. `terraform apply`

**Comment by favoretti** [comment]
Hi there and thank you for reporting this. From what I can see this is what the API returns. Can you please elaborate on why you want to see just GUID and not a fully-qualified ID there? Also, from the looks of it `role_definition_id` is an argument and not an attribute despite being of type `Computed` and is used as an alternative to `name` argument to retrieve the role assignment.

**Comment by Shr3ps** [comment]
Hello @favoretti , a little up on this issue which is still present.
In fact, the example from the documentation here https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lighthouse_definition doesn't work out of the box.

`role_definition_id     = data.azurerm_role_definition.contributor.role_definition_id` wants the Role Definition GUID only and not the ID (full path).

**Comment by Shr3ps** [comment]
An idea might be to introduce maybe a new attribute `role_definition_uid` for this?

**Comment by samuelschnurr** [comment]
> Hi there and thank you for reporting this. From what I can see this is what the API returns. Can you please elaborate on why you want to see just GUID and not a fully-qualified ID there? Also, from the looks of it `role_definition_id` is an argument and not an attribute despite being of type `Computed` and is used as an alternative to `name` argument to retrieve the role assignment.

For example you need just the GUID in this resource:

azurerm_monitor_action_group -> arm_role_receiver -> role_id

My workaround is currently the ugly split: 

```
data "azurerm_role_definition" "monitoringcontributor" {
  name  = "Monitoring Contributor"
  scope = data.azurerm_subscription.current.id
}

locals {
  splitArray              = split("/", data.azurerm_role_definition.monitoringcontributor.id)
  monitoringContributorId = element(local.splitArray, length(local.splitArray) - 1)
}
[...]
role_id                 = local.monitoringContributorId
```

**Comment by catriona-m** [comment]
Thanks for reporting this @BzSpi - it looks like this issue has been fixed by #24320 so this can now be closed. Thanks!
